mesonbuild/options.py

```diff
@@ -192,6 +193,7 @@ def __repr__(self) -> str:
         return f'OptionKey({self.name!r}, {self.subproject!r}, {self.machine!r})'
 
     @classmethod
+    @lru_cache(None)
```

Comment:
I don't think this has a big impact. It is probably only used for parsing options given by the command line.

Reply:
This gets used in a surprising number of places, many of them because we still are doing user input validation below the interpreter.